### PR TITLE
Give away point

### LINKED ISSUE #50: give away x point to arbitarary user
code content of updateNft

### PR DESCRIPTION
**Pull request by geeknees**
- add giveAwayPoint function / spec
- fix script

FIx #50
```
👺 yarn run test
yarn run v1.22.17
$ npx hardhat test


  PodCastNFT
    giveAwayPoint
      ✔ revert without nft
      ✔ has 0 points at first
      ✔ can grant 100 points
      ✔ can grant 0 points


  4 passing (1s)

✨  Done in 1.76s.

```

### CODE REVIEW

File: test/mint.js [hunk 1]
```diff
@@ -106,31 +105,32 @@ describe('PodCastNFT', function () {
...
-  })
-
-  it('cannot mint twice for the same user(owner)', async function () {
-    const mintTx = await contract.mint(
-      'https://example.com/podcast.png',
-      ['Podcast Contributor'],
-      alice.address
-    )
-    await mintTx.wait()
+  describe('mint', () => {
```
Comment (geeknees) [inline test/mint.js:108]: just add `describe`

File: scripts/deploy.js [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
-const hre = require("hardhat");
+const hre = require('hardhat')
 
 const main = async () => {
-  const podcastNFTContract = await hre.ethers.getContractFactory("PodCastNFT");
-  const contract = await podcastNFTContract.deploy([process.env.ADMIN_ADDRESS], false);
+  const podcastNFTContract = await hre.ethers.getContractFactory('PodCastNFT')
+  const contract = await podcastNFTContract.deploy(
+    process.env.TOKEN_ADDRESS,
+    process.env.ADMIN_ADDRESS
```
Comment (SamuraiT) [inline scripts/deploy.js:9]: [should] to avoid confusion, name should be FUND_ADDRESS. 
FUND_ADDRESS should be controlled by only multisig

**Review by SamuraiT — APPROVED** [review]
LGTM with a comment

**Comment by SamuraiT** [comment]
can you also check the contract size ?

**Comment by geeknees** [comment]
Thank you!
I think the contract size will be fine.

before
```
Compiled 1 Solidity file successfully
 ·-----------------|-------------|---------------·
 |  Contract Name  ·  Size (KB)  ·  Change (KB)  │
 ··················|·············|················
 |  console        ·      0.086  ·               │
 ··················|·············|················
 |  Base64         ·      0.086  ·               │
 ··················|·············|················
 |  Counters       ·      0.086  ·               │
 ··················|·············|················
 |  Strings        ·      0.086  ·               │
 ··················|·············|················
 |  Address        ·      0.086  ·               │
 ··················|·············|················
 |  MockERC20      ·      4.921  ·               │
 ··················|·············|················
 |  ERC20          ·      4.921  ·               │
 ··················|·············|················
 |  ERC721         ·      9.159  ·               │
 ··················|·············|················
 |  PodCastNFT     ·     20.545  ·       -0.334  │
 ·-----------------|-------------|---------------·
✨  Done in 6.41s.

```

after
```
Compiled 1 Solidity file successfully
 ·-----------------|-------------|---------------·
 |  Contract Name  ·  Size (KB)  ·  Change (KB)  │
 ··················|·············|················
 |  console        ·      0.086  ·               │
 ··················|·············|················
 |  Base64         ·      0.086  ·               │
 ··················|·············|················
 |  Address        ·      0.086  ·               │
 ··················|·············|················
 |  Counters       ·      0.086  ·               │
 ··················|·············|················
 |  Strings        ·      0.086  ·               │
 ··················|·············|················
 |  MockERC20      ·      4.921  ·               │
 ··················|·············|················
 |  ERC20          ·      4.921  ·               │
 ··················|·············|················
 |  ERC721         ·      9.159  ·               │
 ··················|·············|················
 |  PodCastNFT     ·     20.879  ·       +0.334  │
 ·-----------------|-------------|---------------·
✨  Done in 6.27s.


```

**Comment by SamuraiT** [comment]
good ! 👍

File: contracts/PodCastNFT.sol [hunk 3]
```diff
@@ -142,6 +142,14 @@ contract PodCastNFT is ERC721, Ownable {
         return _mint(_tokenURI, _roles, _to);
     }
 
+    function giveAwayPoint(address _to, uint256 _point)
+        public
+        onlyOwner
+        onlyHolder(_to)
+    {
+        userAttribute[_to].point += _point;
```
Comment (SamuraiT) [inline contracts/PodCastNFT.sol:150]: uh also you need to add claimable token

**Pull request merged**